async-activity/workflow.go

```diff
@@ -0,0 +1,54 @@
+package asyncactivity
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/workflow"
+)
+
+// AsyncActivityWorkflow is a workflow definition starting two activities
+// asynchronously.
+func AsyncActivityWorkflow(ctx workflow.Context, name string) (string, error) {
+	ao := workflow.ActivityOptions{
+		StartToCloseTimeout: 10 * time.Second,
+	}
+	ctx = workflow.WithActivityOptions(ctx, ao)
+
+	// Start activities asynchronously.
```

Comment:
Alright, I see.  This indeed wasn't meant to be async completion --  I was going through the README and I saw: 

---
### Pending examples
Mostly examples we haven't yet ported from https://github.com/temporalio/samples-java/

Async activity calling: Example to be completed

---

which I thought meant essentially porting [this code](https://github.com/temporalio/samples-java/blob/main/core/src/main/java/io/temporal/samples/hello/HelloAsync.java) from the java example:

```java
    public String getGreeting(String name) {

      /*
       * This is our workflow method. We invoke the composeGreeting method two times using
       * {@link io.temporal.workflow.Async#function(Func)}.
       * The results of each async activity method invocation returns us a
       * {@link io.temporal.workflow.Promise} which is similar to a Java {@link java.util.concurrent.Future}
       */
      Promise<String> hello = Async.function(activities::composeGreeting, "Hello", name);
      Promise<String> bye = Async.function(activities::composeGreeting, "Bye", name);

      // After calling the two activity methods async, we block until we receive their results
      return hello.get() + "\n" + bye.get();
    }
```

I suppose since in Java there is also a different syntax which allows calling an activity and waiting for it to complete synchronously, it helps more to have both examples. But given that in Go you always get a Future when calling `ExecuteActivity`, not sure if having this example as well adds as much value as in the java repository (you could pretty much use the hello example and build upon that in this case). I made this PR pretty much because it was on the readme.

> though I would recommend using a selector to wait for first complete since that's a better demonstration for most users

That would end up making this example more or less like `pickfirst`, so I guess in that case I'd be questioning the purpose of this sample.

In that case, I'd suggest dropping the sample, but maybe also removing that section from the README, since there is the pickfirst sample?

Reply:
Yeah, removing that whole "pending" section of the README may be best. @Quinn-With-Two-Ns - thoughts?

Reply:
Yeah I would just remove the "pending" from the README

Reply:
Since that's quite different from what I started with, I made #343 as a result to remove the "pending" and will close this PR afterwards.